src/auth/auth.cpp

```diff
@@ -389,11 +401,24 @@ std::optional<UserOrRole> Auth::SSOAuthenticate(const std::string &scheme,
 }
 
 void Auth::LinkUser(User &user) const {
-  auto link = storage_.Get(kLinkPrefix + user.username());
-  if (link) {
-    auto role = GetRole(*link);
-    if (role) {
-      user.SetRole(*role);
+  // Role
+  {
+    auto link = storage_.Get(kRoleLinkPrefix + user.username());
```

Comment:
I think connecting strings using fmt is better than using + to avoid intermediate string allocations